taichi/transforms/compile_to_offloads.cpp

```diff
@@ -73,6 +73,10 @@ void compile_to_offloads(IRNode *ir,
     irpass::analysis::verify(ir);
   }
 
+  irpass::extract_constant(ir);
```

Comment:
Quick question: If we extract constants before offloading, does that lead to more usage of `GlobalTmp`? Or we have global temp store forwarding that fixes this?

Reply:
I think #812 fixes this.

Reply:
(Although #812 may introduce more copies of constants...)

This pass reduces the number of statements from 83.059% to 83.052%.

Reply:
Sounds good :-) Feel free to merge once CI passes.